src/gui-qt4/libs/seiscomp3/gui/core/connectiondialog.cpp

```diff
@@ -217,7 +217,7 @@ bool ConnectionDialog::fetchDatabase() {
 		Util::StopWatch timer;
 		while ( timer.elapsed() < TimeSpan(5.0) ) {
 			(*_connection)->poll();
-			while ( msg = (*_connection)->readMessage(false) ) {
+			while ( msg == (*_connection)->readMessage(false) ) {
```

Comment:
That is definitely wrong and will break the code. What is meant here is:

```c++
while ( (msg = (*_connection)->readMessage(false)) != NULL )
```

or more readable

```c++
while ( true ) {
  msg = (*_connection)->readMessage(false);
  if ( msg == NULL ) break;
  ...
}
```

Did clang raise an error or just a warning?